torchgeo/datasets/geonrw.py

```diff
@@ -0,0 +1,334 @@
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        to_tensor = transforms.ToTensor()
+
+        path: str = self.file_list[index]
+        utm_coords = os.path.basename(path).split('_')[:2]
+        base_dir = os.path.dirname(path)
+
+        sample: dict[str, Tensor] = {}
+        for modality in self.modalities:
+            modality_path = os.path.join(
+                base_dir, self.modality_filenames[modality](utm_coords)
+            )
+            sample[modality] = to_tensor(self.readers[modality](modality_path))
+
+        # rename to torchgeo standard keys
+        sample['image'] = sample.pop('rgb').float()
+        sample['mask'] = sample.pop('seg').long()
+
+        if self.transforms:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        # check if city names directories exist
+        all_exist = all(
+            os.path.exists(os.path.join(self.root, cn)) for cn in self.city_names
+        )
+        if all_exist:
+            return
+
+        # Check if the tar file has been downloaded
+        if os.path.exists(os.path.join(self.root, self.filename)):
+            extract_archive(os.path.join(self.root, self.filename), self.root)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        # Download the dataset
+        self._download()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        download_and_extract_archive(
+            self.url.format(self.filename),
+            download_root=self.root,
+            md5=self.md5 if self.checksum else None,
+        )
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: str | None = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional suptitle to use for figure
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        showing_predictions = 'prediction' in sample
+        ncols = 3
+        if showing_predictions:
+            prediction = sample['prediction'].long()
+            ncols += 1
+
+        fig, axs = plt.subplots(
+            nrows=1, ncols=ncols, figsize=(ncols * 5, 10), sharex=True
+        )
+
+        axs[0].imshow(sample['image'].permute(1, 2, 0))
+        axs[0].axis('off')
+        axs[1].imshow(sample['dem'].squeeze(0), cmap='gray')
+        axs[1].axis('off')
+        axs[2].imshow(sample['mask'].squeeze(0), cmap=self.colormap)
```

Comment:
See if this helps:
```suggestion
        axs[2].imshow(sample['mask'].squeeze(0), cmap=self.colormap, vmin=0, vmax=10, interpolation='none')
```
This is what we used in L7 Irish/L8 Biome.